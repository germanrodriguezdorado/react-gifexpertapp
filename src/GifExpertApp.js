import { useState } from "react";
import AddCategory from "./componentes/AddCategory";
import GifGrid from "./componentes/GifGrid";

const GifExpertApp = ({ defaultCategories = []}) => {

    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }></AddCategory>
            <hr/>
            <ol>
                {
                    categories.map( category => (
                            <GifGrid 
                                key={ category }
                                category={ category }
                            />
                    ))
                }
            </ol>
            
        </>
    )
}



export default GifExpertApp;