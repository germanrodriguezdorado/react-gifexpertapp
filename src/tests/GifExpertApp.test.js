import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe('Pruebas inherentes al componente <GifExpertApp/>', () => {
    
    test('El componente se debe generar correctamente', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
       
        const categories = ["Marvel", "Hulk"];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);


    });
    
    


});
