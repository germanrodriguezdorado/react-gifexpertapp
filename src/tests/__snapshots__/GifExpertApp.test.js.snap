// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas inherentes al componente <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
    <GifGrid
      category="Hulk"
      key="Hulk"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas inherentes al componente <GifExpertApp/> El componente se debe generar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
